core/metacling/src/TClingRdictModuleFileExtension.h

```diff
@@ -0,0 +1,79 @@
+/// \file TClingRdictModuleFileExtension.h
+///
+/// \brief The file contains facilities to work with C++ module files extensions
+///        used to store rdict files.
+///
+/// \author Vassil Vassilev <vvasilev@cern.ch>
+///
+/// \date May, 2019
+///
+/*************************************************************************
+ * Copyright (C) 1995-2018, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#ifndef ROOT_CLING_RDICT_MODULE_FILE_EXTENSION_H
+#define ROOT_CLING_RDICT_MODULE_FILE_EXTENSION_H
+
+#include "clang/Serialization/ModuleFileExtension.h"
+#include "clang/Basic/LLVM.h"
+#include "llvm/ADT/StringRef.h"
+#include "llvm/Bitcode/BitstreamReader.h"
+#include "clang/Serialization/ASTReader.h"
+#include "clang/Serialization/Module.h"
+#include <string>
+
+/// Rdict module extension block name.
+const std::string ROOT_CLING_RDICT_BLOCK_NAME = "root.cling.rdict";
```

Comment:
Shouldn't those 3 be constexpr?  Could they be inside a namespace or inside TClingRdictModuleFileExtension?

Reply:
Constexpr tells the compiler that the 'content' can be evaluated at compile-time but does not tell they can't change. Here we mean that these do not change. Did you mean making it `const constexpr`? If so we can make the other two like this (we cannot make an std::string constexpr) although what would be the benefit? const is a stronger constraint than constexpr.

Reply:
> constexpr ... but does not tell they can't change.

Humm ...  when can a constexpr change?

Actually that remind me that this will lead to a linking error if this header file is included in more than one compilation unit (because the string is neither static nor extern ...).   This would not be the case if this was a 'constexpr const char *'  [the constexpr applies to the pointer while the const applies to the content]

Reply:
That's a good point. I've moved them in the implementation and made constexpr where possible.